File: src/main/java/com/artmaguire/Person.java
package com.artmaguire;

import java.util.Objects;

public class Person implements Comparable<Person> {
    private String firstName;
    private String lastName;
    private int age;
    private Gender gender;
    private String weapon;
    private District district;
    private int kills = 0;

    public enum Gender {
        Female, Male
    }

    public Person(String firstName, String lastName, int age, Gender gender, District district) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
        this.gender = gender;
        this.district = district;
    }

    public String getName() {
        return firstName + " " + lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public Gender getGender() {
        return gender;
    }

    public void setGender(Gender gender) {
        this.gender = gender;
    }

    public District getDistrict() {
        return district;
    }

    public void setDistrict(District district) {
        this.district = district;
    }

    public String getWeapon() {
        return weapon;
    }

    public void setWeapon(String weapon) {
        this.weapon = weapon;
    }

    public int getKills() {
        return kills;
    }

    public void addKill() {
        this.kills++;
    }

    public void printInfo() {
        System.out.println("Player Name: " + getName());
        System.out.println("Player Age: " + age);
        System.out.println("Player District: " + district);
    }

    @Override
    public int compareTo(Person o) {
        return o.getKills() - this.kills;
    }

    @Override
    public int hashCode() {
        return Objects.hash(firstName, lastName, age, gender, district);
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }
        final Person other = (Person) obj;
        return Objects.equals(this.firstName, other.firstName)
                && Objects.equals(this.lastName, other.lastName)
                && Objects.equals(this.age, other.age)
                && Objects.equals(this.gender, other.gender)
                && Objects.equals(this.district, other.district);
    }
}

File: web/js/Person.js
function Person(districtNumber, gender) {
    this.dNum = districtNumber;
    this.gender = gender;
    this.alive = true;
    this.kills = 0;
    this.statusID = '';
}

Person.Gender = Object.freeze({Female: "Female", Male: "Male"}); //gender enum
Person.Action = Object.freeze({AddKill: "AddKill", Kill: "Kill"}); //gender enum

Person.prototype.isAlive = function() {
    return this.alive;
};

Person.prototype.act = function(action) {
    switch(action) {
        case Person.Action.AddKill: { this.addKill(); break; }
        case Person.Action.Kill: { this.kill(); break; }
    }
};

Person.prototype.getGender = function() {
    return this.gender;
};

Person.prototype.getKills = function() {
    return this.kills;
};

Person.prototype.addKill = function() {
    this.kills++;
    $( '#' + this.statusID + 'Kills' ).text(this.kills.toString() + ((this.kills === 1) ? ' Kill' : ' Kills'));
};

Person.prototype.setStatusID = function(statusID) {
    this.statusID = statusID;
};

Person.prototype.kill = function() {
    this.alive = false;
    $( '#' + this.statusID ).detach().appendTo( '#deadPeople' );
};

Person.prototype.getFirstName = function() {
    return $( '#firstName' + this.gender + this.dNum).val();
};

Person.prototype.setFirstName = function(firstName) {
    $( '#firstName' + this.gender + this.dNum).val(firstName);
};

Person.prototype.getLastName = function() {
    return $( '#lastName' + this.gender + this.dNum).val();
};

Person.prototype.setLastName = function(lastName) {
    $( '#lastName' + this.gender + this.dNum).val(lastName);
};

Person.prototype.getName = function() {
    return this.getFirstName() + ' ' + this.getLastName();
};

Person.prototype.getAge = function() {
    return $( '#age' + this.gender + this.dNum).val();
};

Person.prototype.setAge = function(age) {
    $( '#age' + this.gender + this.dNum).val(age);
};
